subsys/logging/backends/log_backend_net.c

```diff
@@ -48,12 +46,33 @@ static int line_out(uint8_t *data, size_t length, void *output_ctx)
 {
 	struct log_backend_net_ctx *ctx = (struct log_backend_net_ctx *)output_ctx;
 	int ret = -ENOMEM;
+	struct msghdr msg = { 0 };
+	struct iovec io_vector[2];
+	int pos = 0;
 
 	if (ctx == NULL) {
 		return length;
 	}
 
-	ret = zsock_send(ctx->sock, data, length, ZSOCK_MSG_DONTWAIT);
+#if defined(CONFIG_NET_TCP)
+	char len[sizeof("123456789")];
+
+	if (ctx->is_tcp) {
+		(void)snprintk(len, sizeof(len), "%zu ", length);
+		io_vector[pos].iov_base = (void *)len;
+		io_vector[pos].iov_len = strlen(len);
+		pos++;
+	}
+#endif
+
+	io_vector[pos].iov_base = (void *)data;
+	io_vector[pos].iov_len = length;
+	pos++;
+
+	msg.msg_iov = io_vector;
+	msg.msg_iovlen = pos;
+
+	ret = zsock_sendmsg(ctx->sock, &msg, ZSOCK_MSG_DONTWAIT);
```

Comment:
Does it sill make sense to use non-blocking send with TCP? It's no longer best-effort as with UDP, we should keep the stream consistent. And with TCP it's easier to get EAGAIN as a result (by filling TX window).

Reply:
Not sure if we should block here. I mean the logging would get delayed in that case which could be problematic if synchronous logging is used. And if we cannot send, we should drop the message anyway.

Reply:
And wouldn't it be a problem that we send the message length first, and then incomplete message (sorry, I didn't dig into RFC yet). With TCP it's not all or nothing, we could just send part of the buffer.

Reply:
I was trying to avoid few byte send that was happening if I had two sends.